src/ui/src/builder/composables/useDragToSelect.ts

```diff
@@ -0,0 +1,341 @@
+
+		const wrapper = wrapperRef.value;
+		if (!wrapper) return;
+
+		try {
+			const wrapperRect = wrapper.getBoundingClientRect();
+			const currentX = ev.clientX - wrapperRect.left;
+			const currentY = ev.clientY - wrapperRect.top + wrapper.scrollTop;
+
+			const { startX, startY } = selectionRect.value;
+			const left = Math.min(startX, currentX);
+			const top = Math.min(startY, currentY) - wrapper.scrollTop;
+			const width = Math.abs(currentX - startX);
+			const height = Math.abs(currentY - startY);
+
+			selectionRect.value = {
+				...selectionRect.value,
+				left,
+				top,
+				width,
+				height,
+			};
+		} catch {
+			cleanupDragSelection();
+		}
+	}
+
+	function handleDocumentMousemove(ev: MouseEvent) {
+		if (!selectionRect.value.isSelecting) return;
+		updateSelectionRect(ev);
+		ev.preventDefault();
+	}
+
+	function handleMousedown(ev: MouseEvent) {
+		if (builderMode.value === "preview") return;
+		if (isAnnotating.value) return;
+
+		if (ev.shiftKey || ev.ctrlKey || ev.metaKey) return;
+
+		if (builderMode.value === "blueprints" && !ev.altKey) return;
+
+		const target = ev.target as HTMLElement;
+
+		const targetEl = target.closest<HTMLElement>("[data-writer-id]");
+		if (targetEl) {
+			if (builderMode.value === "blueprints") {
+				if (!shouldAllowSelectionInBlueprints(target)) return;
+			} else {
+				return;
+			}
+		}
+
+		const unselectableEl = target.closest<HTMLElement>(
+			"[data-writer-unselectable]",
+		);
+		if (unselectableEl) return;
+
+		if (target.classList.contains("selectionRectangle")) return;
+
+		const wrapper = wrapperRef.value;
+		if (!wrapper) return;
+
+		if (!wrapper.contains(target)) return;
+
+		if (builderMode.value === "blueprints") {
+			if (!isClickOnBlueprintsCanvas(target)) return;
+		}
+
+		const wrapperRect = wrapper.getBoundingClientRect();
+		const startX = ev.clientX - wrapperRect.left;
+		const startY = ev.clientY - wrapperRect.top + wrapper.scrollTop;
+
+		selectionRect.value = {
+			isSelecting: true,
+			startX,
+			startY,
+			left: startX,
+			top: startY - wrapper.scrollTop,
+			width: 0,
+			height: 0,
+		};
+
+		document.body.style.userSelect = "none";
+		document.body.style.cursor = "crosshair";
+
+		document.addEventListener("mousemove", handleDocumentMousemove, {
+			signal: dragAbortController.signal,
+		});
+
+		ev.stopPropagation();
+		ev.preventDefault();
+	}
+
+	function handleMousemove(ev: MouseEvent) {
+		if (selectionRect.value.isSelecting) {
+			if (builderMode.value === "blueprints") {
+				ev.stopPropagation();
+			}
+			updateSelectionRect(ev);
+			ev.preventDefault();
+			return;
+		}
+
+		if (builderMode.value === "preview") return;
+		if (isAnnotating.value) return;
+
+		if (ev.shiftKey || ev.ctrlKey || ev.metaKey) {
+			isHoveringSelectableArea.value = false;
+			return;
+		}
+
+		if (builderMode.value === "blueprints" && !ev.altKey) {
+			isHoveringSelectableArea.value = false;
+			return;
+		}
+
+		const target = ev.target as HTMLElement;
```

Comment:
It's better to cast at runtime.

```suggestion
    if (!(ev.target instanceof HTMLElement)) return
		const target = ev.target;
```